sdk/eventhub/Azure.Messaging.EventHubs/src/Core/TransportProducerPool.cs

```diff
@@ -175,17 +167,18 @@ internal TransportProducerPool()
         ///
         public virtual async Task CloseAsync(CancellationToken cancellationToken = default)
         {
-            await EventHubProducer.CloseAsync(cancellationToken).ConfigureAwait(false);
+            ExpirationTimer.Dispose();
```

Comment:
Can this throw?   If so, we should consider wrapping in a try/catch so that we still close the producers.

Reply:
Great point. 

Should we use ```Timer.Dispose(WaitHandle)``` and wait that ```Timer``` has been disposed before trying closing the other producers? 

Not sure it would play nicely with the ```CancellationToken``` passed in as a parameter.

Should we rethrow an exception later? 

If we did not the ```CreateExpirationTimerCallback``` might run again until before  ```TransportProducerPool``` isn't garbage collected?

Reply:
I would say that just wrapping the timer in a try/catch should work:  `try { ExpirationTimer.Dispose(); }  catch {}`  is pretty much all that was in my head.   Since the failure would not impact developers using the library and isn't something that they caused, I don't  see value in surfacing it - especially since we're just cleaning up at this point and we know that it won't corrupt state in a way that would impact future use.

Since the pool is an internal part of cleaning up, I agree that we don't want to check the cancellation token here;  once we get to the point of actually cleaning up resources in the outer class, we need to be sure that all the inner steps run so that we don't end up in an inconsistent state.   The outermost class usually checks the token once before it starts doing any heavy lifting and after that ignores it, as teardown has started.    Long-winded way of saying,  I think we're good here.  :smile:

Reply:
Fantastic, to double-check.

I turned this:

```csharp
foreach (var poolItem in Pool.Values)
{
    pendingCloses.Add(poolItem.PartitionProducer.CloseAsync(CancellationToken.None));
}
```

into this:

```csharp
foreach (var poolItem in Pool.Values)
{
    pendingCloses.Add(poolItem.PartitionProducer.CloseAsync(cancellationToken));
}
```

Not sure actually I touched things I shouldn't have touched 😄

Reply:
At the point of the process that you're in, I would definitely go with `CancellationToken.None` here;  if the token gets signaled while you're tearing down, we don't want to honor it as this would leave things in an inconsistent state.   In this case, you'd have the timer and other bits disposed while the producer would honor the token because it was signaled before the close process begun.